modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2393,6 +2393,17 @@ def apply(self, func, axis, *args, **kwargs):
         else:
             return self._callable_func(func, axis, *args, **kwargs)
 
+    def apply_on_series(self, func, *args, **kwargs):
+        args = try_cast_to_pandas(args)
+        kwargs = try_cast_to_pandas(kwargs)
+
+        assert self.is_series_like()
+        return self.__constructor__(
+            self._modin_frame.apply_full_axis(
+                1, lambda df: df.squeeze(axis=1).apply(func, *args, **kwargs)
```

Comment:
@pyrito please add a comment here which is explaining why we picked `apply_full_axis` instead of `map`